src/throttle/amoc_throttle.erl

```diff
@@ -3,57 +3,63 @@
-         start/4,
-         send/3,
-         send/2,
-         send_and_wait/2,
-         wait/1,
-         run/2,
-         pause/1,
-         resume/1,
-         change_rate/3,
-         change_rate_gradually/6,
-         stop/1]).
-
--deprecated([{send_and_wait, 2, "use wait/1 instead"}]).
-
--define(DEFAULT_NO_PROCESSES, 10).
--define(DEFAULT_INTERVAL, 60000). %% one minute
--define(NONNEG_INT(N), (is_integer(N) andalso N >= 0)).
--define(POS_INT(N), (is_integer(N) andalso N > 0)).
+-export([start/2, stop/1,
+         send/2, send/3, wait/1,
+         run/2, pause/1, resume/1, unlock/1,
+         change_rate/2, change_rate_gradually/2]).
 
 -type name() :: atom().
--type rate() :: pos_integer().
--type interval() :: non_neg_integer().
-%% In milliseconds, defaults to 60000 (one minute) when not given.
-%% An interval of 0 means no delay at all, only the number of simultaneous executions will be
-%% controlled, which corresponds to the number of processes started
--export_type([name/0, rate/0, interval/0]).
-
-%% @see start/4
--spec start(name(), rate()) -> ok | {error, any()}.
-start(Name, Rate) ->
-    start(Name, Rate, ?DEFAULT_INTERVAL).
+%% Atom representing the name of the throttle.
+
+-type rate() :: infinity | non_neg_integer().
+%% Number of events per given `t:interval/0', or
+%% - infinity for effectively unlocking all throttling,
+%% - zero for effectively pausing the throttle.
+
+-type interarrival() :: infinity | non_neg_integer().
+%% Time in milliseconds between two events, or
+%% - infinity for effectively pausing the throttle,
+%% - zero for effectively unlocking all throttling.
+
+-type interval() :: pos_integer().
+%% In milliseconds, defaults to 60000 (one minute).
+
+-type t() :: #{rate := rate(), interval => interval()} |
+             #{interarrival := interarrival()}.
+%% Throttle unit of measurement
+
+-type gradual_rate_config() :: #{from_rate := non_neg_integer(),
+                                 to_rate := non_neg_integer(),
+                                 interval => interval(),
+                                 step_interval => pos_integer(),
+                                 step_size => pos_integer(),
+                                 step_count => pos_integer(),
+                                 duration => pos_integer()} |
+                               #{from_interarrival := pos_integer(),
+                                 to_interarrival := pos_integer(),
+                                 step_interval => pos_integer(),
+                                 step_size => pos_integer(),
+                                 step_count => pos_integer(),
+                                 duration => pos_integer()}.
+%% Configuration for a gradual throttle rate change
+%%
+%% `From' and `To' rates are required. `interval' defaults to 1s and `step_size' to 1 (or -1 if applies),
+%% that is, the throttle will be changed in increments of 1.
+%%
+%% All other values can be calculated from the provided.
```

Comment:
are interval values given in ms? please add the comment.

Reply:
It is commented in line 24, where the interval type is defined. Only thing this isn't using such type, changing that now.

Reply:
also please remove references to non-existing  variables `From` and `To`